content/4_Gold/Combinatorics.mdx

```diff
@@ -240,7 +339,95 @@ long long choose(long long n, long long r, long long p) {
 	return fac[n] * inv[r] % p * inv[n - r] % p;
 }
 ```
+
 </CPPSection>
+<JavaSection>
+
+```java
+import java.util.*;
+
+public class BinomialCoefficients {
+	private static final int MAXN = (int)1e6;
+	private static long[] fac = new long[MAXN + 1];
+	private static long[] inv = new long[MAXN + 1];
+
+	/** Computes x^n modulo m in O(log p) time. */
+	private static long exp(long x, long n, long m) {
```

Comment:
maybe have a link to the mod exp part of the guide? if the c++ doesn't do it ignore this comment